src/panels/lovelace/cards/hui-tile-card.ts

```diff
@@ -0,0 +1,210 @@
+    await import("../editor/config-elements/hui-tile-card-editor");
+    return document.createElement("hui-tile-card-editor");
+  }
+
+  public static getStubConfig(
+    hass: HomeAssistant,
+    entities: string[],
+    entitiesFallback: string[]
+  ): TileCardConfig {
+    const includeDomains = ["sensor", "light", "switch"];
+    const maxEntities = 1;
+    const foundEntities = findEntities(
+      hass,
+      maxEntities,
+      entities,
+      entitiesFallback,
+      includeDomains
+    );
+
+    return {
+      type: "tile",
+      entity: foundEntities[0] || "",
+    };
+  }
+
+  @property({ attribute: false }) public hass?: HomeAssistant;
+
+  @state() private _config?: TileCardConfig;
+
+  public setConfig(config: ThermostatCardConfig): void {
+    if (!config.entity) {
+      throw new Error("Specify an entity");
+    }
+
+    const supportToggle =
+      config.entity && DOMAINS_TOGGLE.has(computeDomain(config.entity));
+
+    this._config = {
+      tap_action: {
+        action: "more-info",
+      },
+      icon_tap_action: {
+        action: supportToggle ? "toggle" : "more-info",
+      },
+      ...config,
+    };
+  }
+
+  public getCardSize(): number {
+    return 1;
+  }
+
+  private _handleAction(ev: ActionHandlerEvent) {
+    handleAction(this, this.hass!, this._config!, ev.detail.action!);
+  }
+
+  private _handleIconAction() {
+    const config = {
+      entity: this._config!.entity,
+      tap_action: this._config!.icon_tap_action,
+    };
+    handleAction(this, this.hass!, config, "tap");
+  }
+
+  render() {
+    if (!this._config || !this.hass) {
+      return html``;
+    }
+    const entityId = this._config.entity;
+    const entity = entityId ? this.hass.states[entityId] : undefined;
+
+    if (!entity) {
+      return html`
+        <ha-card class="disabled">
+          <div class="tile">
+            <ha-tile-icon .iconPath=${mdiHelp}></ha-tile-icon>
+            <ha-tile-info
+              .primary=${entityId}
+              secondary=${this.hass.localize("ui.card.tile.not_found")}
+            ></ha-tile-info>
+          </div>
+        </ha-card>
+      `;
+    }
+
+    const icon = this._config.icon || entity.attributes.icon;
+    const iconPath = stateIconPath(entity);
+
+    const name = this._config.name || entity.attributes.friendly_name;
+    const stateDisplay = computeStateDisplay(
+      this.hass.localize,
+      entity,
+      this.hass.locale
+    );
+
+    const iconStyle = {};
+    if (this._config.color && !STATES_OFF.includes(entity.state)) {
+      iconStyle["--main-color"] = computeRgbColor(this._config.color);
+    }
+
+    return html`
+      <ha-card style=${styleMap(iconStyle)}>
+        <div class="tile">
+          <ha-tile-icon
+            .icon=${icon}
+            .iconPath=${iconPath}
+            role="button"
+            tabindex="0"
+            @action=${this._handleIconAction}
+            .actionHandler=${actionHandler()}
```

Comment:
If we don't pass any config anyway, we can just listen to click events?

Reply:
For now, but I think we should add hold_action for advanced user.